tests/Unit/UtilsTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+namespace Automattic\LegacyRedirector\Tests\Unit;
+
+use Automattic\LegacyRedirector\Utils;
+use Brain\Monkey;
+use Yoast\WPTestUtils\BrainMonkey\TestCase;
+
+/**
+ * Capability Class Unit Test
+ */
+final class UtilsTest extends TestCase {
+
+	/**
+	 * Utils::mb_parse_url test method
+	 *
+	 * @covers \Automattic\LegacyRedirector\Utils::mb_parse_url
+	 */
+	public function test_mb_parse_url() {
+
+		Monkey\Functions\stubs(
+			array(
+				'wp_parse_url' => static function ( $url, $component ) {
+					return parse_url( $url, $component );
+				},
+			)
+		);
+
+		$url = 'https://www.example.org';
+		$this->do_assetion_mb_parse_url( $url, 'https', 'www.example.org', '', '' );
+
+		$url = 'https://www.example.org/';
+		$this->do_assetion_mb_parse_url( $url, 'https', 'www.example.org', '/', '' );
+
+		$url = 'https://www.example.com/test';
+		$this->do_assetion_mb_parse_url( $url, 'https', 'www.example.com', '/test', '' );
+
+		$url = 'http://www.example.com//فوتوغرافيا/?test=فوتوغرافيا';
+		$this->do_assetion_mb_parse_url( $url, 'http', 'www.example.com', '//فوتوغرافيا/', 'test=فوتوغرافيا' );
+
+		$url = '/فوتوغرافيا/?test=فوتوغرافيا';
+		$this->do_assetion_mb_parse_url( $url, '', '', '/فوتوغرافيا/', 'test=فوتوغرافيا' );
+
+		$url = '/فوتوغرافيا/?test2=فوتوغرافيا&test=فوتوغرافيا';
+		$this->do_assetion_mb_parse_url( $url, '', '', '/فوتوغرافيا/', 'test2=فوتوغرافيا&test=فوتوغرافيا' );
```

Comment:
Seems like these values should be held in a dataProvider method?

Reply:
added a new data provider for it.